src/Operations/ConvertContoursToMeshes.cc

```diff
@@ -544,8 +591,178 @@ bool ConvertContoursToMeshes(Drover &DICOM_data,
                         const auto f_C = static_cast<uint64_t>(fs[2] + old_face_count);
                         amesh.faces.emplace_back( std::vector<uint64_t>{{f_A, f_B, f_C}} );
                     }
+
+                }else if( (N_upper == 2) && (N_lower == 1) ){
+                    //check if the upper plane contains enclosed contours
+                    if (projected_contours_overlap(*m_cp_it, pcs.upper.front(),*m_cp_it, pcs.upper.back())
```

Comment:
if you keep checking overlap and not intersect all the time, should probably replace by lambda function